packages/sdk/src/controllers/VariableController.ts

```diff
@@ -620,6 +620,27 @@ export class VariableController {
         return res.getVariablePrivateData(id).then((result) => getEditorResponseData<PrivateData>(result));
     };
 
+    /**
+     * Gets the variables having required values that are not set
+     *
+     * PDF output won't work if all required variables are not set
+     *
+     * @returns the list of invalid variables
+     */
+    getAllRequiredAndInvalid = async () => {
```

Comment:
Since we're blocking output generation for PDF, this is for integrators to be able to list which variable needs to be fixed, this is maybe something they could check themselves but this is not necessarily obvious in all cases, e.g. list

Reply:
I see, have you considered adding a `readonly` property `isSet` (or something else) on `Variable`?

Reply:
No and this is a good idea.